### ISSUE TITLE
Does not work with Sharp PN-M501

### ISSUE THREAD

**Issue by Dwvsplat** [bitfocus/companion-module-sharp-tv#3]
None of these commands work with the sharp PN-M501. I think they are close.
Here is the command manual:
[pnb401-pnb501-pnm401-pnm501_us1_manual_en.pdf](https://github.com/bitfocus/companion-module-sharp-tv/files/5990737/pnb401-pnb501-pnm401-pnm501_us1_manual_en.pdf)

**Comment by McHauge** [comment]
@Dwvsplat Does that TV have ethernet? Companion itself, does not support RS232 or other serial based protocols, and requeres an IP connection to the units it contol.

**Comment by Dwvsplat** [comment]
@McHauge yes the tv does have ethernet (last page of the pdf). It uses the same commands as RS232. I can connect to it with the sharp module & it says connected ( I can also change the port and get errors), so I believe it is using a different code than the PN-LE901 (and/or this current module). All I really need is on/off & input commands.

**Comment by McHauge** [comment]
@Dwvsplat well it's the same commands, did you turn "TCP Server ON" on the TV itself, and are your user ID and password correct? 
![image](https://user-images.githubusercontent.com/31018035/108206353-98bab180-7126-11eb-96d8-691b180ae3f6.png)

And if you get an Error what does it say?

**Comment by McHauge** [comment]
another note, if the TV has any energy saving featues enabled, then it will not receve any commands. all power saving (energy star ect.) needs to be turned off

**Comment by Dwvsplat** [comment]
Yes energy star is off. 
I enabled lan and turned on telnet service. 
I am not using a username or password
The only error I get is if I change the port (and don't change it in companion).

**Comment by McHauge** [comment]
well you need to use the user ID and password for the module to work, i't requered by the TV

**Comment by McHauge** [comment]
it will connect without user ID and password, that is correct, but it wont do anything, as the first thing it does is "login" and without it the commands will just be ingored and fall to the ground.

**Comment by Dwvsplat** [comment]
So you think I NEED to put in a username and password?
I know for a fact I do not have to do that with the PN-LE901 (this can actually cause issues on that tv).

**Comment by McHauge** [comment]
Well i'm just reading the documentation, and it says that you have to, I don't own or have acces to any Sharp TV for testing, the whole thing is based soely on documentation, as the original requester never returned with an answer to if it worked or not

From Page 9 in the dokumentation:  
![image](https://user-images.githubusercontent.com/31018035/108210898-614f0380-712c-11eb-94ec-b3472395ecd6.png)

**Comment by McHauge** [comment]
But yes just looked at the PN-LE901 documentation, and that TV does not use a User id or Password, so it seems it deppends on the model, but the PN-M501 seems to need it based on the documentation

From the PN-LE901's docs it seems to be optinal:
![image](https://user-images.githubusercontent.com/31018035/108213102-f6eb9280-712e-11eb-98a7-1d09a6d3ef98.png)

**Comment by Dwvsplat** [comment]
I am at the tv. I put in a username (1234) and password (1234). 
I was getting a network error warning from companion....utill I put in the username and password (connects perfectly).
Checked to see if energy save is off (it was).
Tried power off command....nothing.
Tried input command...nothing
Tried mute command... nothing
Ran a ping.....I can ping it fine
Can log into it with http.

**Comment by McHauge** [comment]
well, what happens if you do it manually, with a TCP client and connects to the tv and sends the commands directly from there: POWR1\r
POWR   1\r
INPS0\r
INPS   0\r

**Comment by McHauge** [comment]
[Packet Sender](https://packetsender.com/) Is what I personaly use for TCP testing.

**Comment by Dwvsplat** [comment]
This as what I was going to do in terminal (mac)..Correct?

echo -e -n '1234\x0d1234\x0dIPOWR1\r   \x0d' | \
        socat - tcp4:192.168.120.60:10007

**Comment by McHauge** [comment]
no idea how that commands works, I was interested in both if it reacted but also what it is responding just by connecting. it is, after all, a bidirectional connection and not just a request.

also from what I can gather you are sending the CR twice with both \r and \x0d
